eng/mark-shipped.ps1

```diff
@@ -0,0 +1,51 @@
+[CmdletBinding(PositionalBinding=$false)]
+param ()
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function MarkShipped([string]$dir) {
+    $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
+    $shipped = Get-Content $shippedFilePath
+    if ($null -eq $shipped) {
+        $shipped = @()
+    }
+
+    $unshippedFilePath = Join-Path $dir "PublicAPI.Unshipped.txt"
+    $unshipped = Get-Content $unshippedFilePath
+    $removed = @()
+    $removedPrefix = "*REMOVED*";
+    Write-Host "Processing $dir"
+
+    foreach ($item in $unshipped) {
+        if ($item.Length -gt 0) {
+            if ($item.StartsWith($removedPrefix)) {
+                $item = $item.Substring($removedPrefix.Length)
+                $removed += $item
+            }
+            elseif ($item -ne "#nullable enable") {
+                $shipped += $item
+            }
+        }
+    }
+
+    $shipped | Sort-Object | Where-Object { -not $removed.Contains($_) } | Out-File $shippedFilePath -Encoding Ascii
+    "" | Out-File $unshippedFilePath -Encoding Ascii
+}
+
+try {
+    Push-Location "$PSScriptRoot\..\src"
+
+    foreach ($file in Get-ChildItem -re -in "PublicApi.Shipped.txt") {
```

Comment:
don't use the -re and -in shortcuts, spell out the parameter names. 

you might also want to avoid the whole relative path thing and specify path directly. Ideally using * globs. If you know that all your public api files are in src/*/PublicApi/PublicApi.Shipped.txt" or something like that. So you don't have to recurse the whole repo and then filter.